test_rclcpp/test/test_allocator.cpp

```diff
@@ -292,15 +304,91 @@ constexpr bool operator!=(const InstrumentedAllocator<T> &,
   return false;
 }
 
-const size_t iterations = 1;
 
 using rclcpp::memory_strategies::allocator_memory_strategy::AllocatorMemoryStrategy;
-using rclcpp::message_memory_strategy::MessageMemoryStrategy;
 
-TEST(CLASSNAME(test_allocator, RMW_IMPLEMENTATION), shared_ptr) {
-  rclcpp::init(0, nullptr);
-  auto node = rclcpp::Node::make_shared("test_allocator_shared_ptr", false);
+class AllocatorTestFixture : public::testing::Test
+{
+protected:
+  std::string test_name_;
+
+  rclcpp::node::Node::SharedPtr node_;
+  rclcpp::executors::SingleThreadedExecutor::SharedPtr executor_;
+  rclcpp::memory_strategy::MemoryStrategy::SharedPtr memory_strategy_;
+  rclcpp::publisher::Publisher<test_rclcpp::msg::UInt32,
+  InstrumentedAllocator<void>>::SharedPtr publisher_;
+  rclcpp::message_memory_strategy::MessageMemoryStrategy<test_rclcpp::msg::UInt32,
+  InstrumentedAllocator<void>>::SharedPtr msg_memory_strategy_;
+  std::shared_ptr<InstrumentedAllocator<void>> alloc;
+
+  bool intra_process_;
+
+  size_t initialization_allocs_;
+  size_t initialization_deallocs_;
+
+  using UInt32Allocator = InstrumentedAllocator<test_rclcpp::msg::UInt32>;
+  using UInt32Deleter = rclcpp::allocator::Deleter<UInt32Allocator, test_rclcpp::msg::UInt32>;
+
+  void initialize(bool intra_process, std::string name)
+  {
+    test_name_ = name;
+    intra_process_ = intra_process;
+
+    auto context = rclcpp::contexts::default_context::get_global_default_context();
+    auto intra_process_manager_state =
+      std::make_shared<rclcpp::intra_process_manager::IntraProcessManagerState<UInt32Allocator>>();
+    context->get_sub_context<rclcpp::intra_process_manager::IntraProcessManager>(
+      intra_process_manager_state);
+
+    node_ = rclcpp::Node::make_shared(name, context, intra_process);
+    alloc = std::make_shared<InstrumentedAllocator<void>>();
+    msg_memory_strategy_ =
+      std::make_shared<rclcpp::message_memory_strategy::MessageMemoryStrategy
+      <test_rclcpp::msg::UInt32,
+      InstrumentedAllocator<void>>>(alloc);
+    publisher_ = node_->create_publisher<test_rclcpp::msg::UInt32>(name, 10, alloc);
+    memory_strategy_ =
+      std::make_shared<AllocatorMemoryStrategy<InstrumentedAllocator<void>>>(alloc);
+    executor_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>(memory_strategy_);
+
+    executor_->add_node(node_);
+  }
+
+  AllocatorTestFixture()
+  {
+    global_allocs = 0;
+    global_runtime_allocs = 0;
+    global_runtime_deallocs = 0;
+    global_deallocs = 0;
+    num_allocs = 0;
+    num_deallocs = 0;
```

Comment:
I believe an initializer list won't work here because these are global static variables.

They are static global variables because they are shared by the test fixture, all instances of the allocator class, and the static global member function "check_stacktrace".

However they get set and reset back to zero when each test case is initialized/deinitialized.

Alternatively, I could make these static member variables of the Allocator and declare a friend relationship between the Allocator and the test fixture.